emcc.py

```diff
@@ -1293,12 +1296,6 @@ def has_c_source(args):
         newargs.append(f)
         add_link_flag(len(newargs), f)
 
-    if not shared.Settings.STRICT:
-      # The preprocessor define EMSCRIPTEN is deprecated. Don't pass it to code
-      # in strict mode. Code should use the define __EMSCRIPTEN__ instead.
-      newargs += ['-DEMSCRIPTEN']
-
-    newargs = shared.COMPILER_OPTS + shared.get_cflags(newargs) + newargs
```

Comment:
I read through this but I have not been able to convince myself that this change has no effects. The order changes at minimum I think? And looks like newargs will be missing some things it had before, like the output from `get_cflags`?

Reply:
This change basically only injects `shared.COMPILER_OPTS + shared.get_cflags(newargs)` at the beginning of the argument list if we are actually compiling C/C++.

If we are compiling assembly we don't want those flags.     Hopefully now that I've re-based on top of some cleanups this change is much more readable.

Reply:
I see, thanks!

Perhaps a comment in the source? It wasn't obvious to me.